analyses/filter-mutation-frequencies-tables/filter-mutation-frequencies-tables.Rmd

```diff
@@ -0,0 +1,324 @@
+  readr::write_tsv(file.path(results_dir, file_name))
+
+  # Write filtered JSON file
+  json_file <- paste(unlist(str_split(file_name, "\\."))[1], "json", sep = ".")
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+    jsonlite::write_json(file.path(results_dir,  json_file))  
+
+return(freq_table)
+}
+```
+
+#### Gene level SNV frequencies
+```{r}
+# Remove Ensembl gene identifiers
+gene_level_snv <- filter_freq_table(
+  file.path(snv_results_dir, 
+            "gene-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 
+
+# Ensembl gene identifiers removed
+filtered_ensg <- gene_level_snv %>% 
+  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
+  dplyr::select(Gene_symbol, targetFromSourceId) %>%
+  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
+  dplyr::distinct()
+
+# read the SNV consensus file
+consensus_ensg <- data.table::fread(
+  file.path(data_dir, "snv-consensus-plus-hotspots.maf.tsv.gz"), 
+  select = c("Tumor_Sample_Barcode", "Gene"), showProgress = FALSE) %>% 
+  tibble() %>% 
+  dplyr::filter(!is.na(Gene)) %>% 
+  dplyr::rename(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode) %>% 
+  dplyr::distinct()
+
+# merge sample IDs to the ENSG IDs not in gencode v38
+filtered_ensg <- filtered_ensg %>% 
+  dplyr::left_join(consensus_ensg, by=c("Ensembl_ID" = "Gene"))
+
+# merge OpenPedCan cohorts to the ENSG IDs not in gencode v38
+filtered_ensg <- filtered_ensg %>% 
+  dplyr::left_join(sample_ids, by= "Kids_First_Biospecimen_ID")
+
+# write to file
+readr::write_tsv(filtered_ensg,
+  file.path(results_dir, 
+            "gene-level-snv-consensus-annotated-mut-freq_dropped_ensg.tsv.gz"))
+
+# Display table
+filtered_ensg %>% dplyr::select(Gene_symbol, Ensembl_ID, cohort) %>% 
+  dplyr::distinct()
+
+# Remove frequencies data frame
+rm(gene_level_snv, filtered_ensg)
+```
+
+#### Variant level SNV frequencies
+```{r}
+# Remove Ensembl gene identifiers
+variant_level_snv <- filter_freq_table(
+  file.path(snv_results_dir, 
+            "variant-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 
+
+# Ensembl gene identifiers removed
+filtered_ensg <- variant_level_snv %>% 
+  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
+  dplyr::select(Gene_symbol, targetFromSourceId) %>%
+  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
+  dplyr::distinct()
+
+# merge sample IDs to the ENSG IDs not in gencode v38
+filtered_ensg <- filtered_ensg %>% 
+  dplyr::left_join(consensus_ensg, by=c("Ensembl_ID" = "Gene"))
+
+# merge OpenPedCan cohorts to the ENSG IDs not in gencode v38
+filtered_ensg <- filtered_ensg %>% 
+  dplyr::left_join(sample_ids, by= "Kids_First_Biospecimen_ID")
+
+# write to file
+readr::write_tsv(filtered_ensg,
+  file.path(results_dir, 
+            "variant-level-snv-consensus-annotated-mut-freq_dropped_ensg.tsv.gz"))
+
+# Display table
+filtered_ensg %>% dplyr::select(Gene_symbol, Ensembl_ID, cohort) %>% 
+  dplyr::distinct()
```

Comment:
I am actually thinking to get the number of samples in each cohort - so could you please modify to:
```
filtered_ensg %>% dplyr::select( Ensembl_ID, cohort) %>% 
  table()
# display the samples that have the ENSEMBL ID that are in PMTL list
filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_df$Ensembl_ID) %>% 
select(Ensemble_ID, Kids_First_Biospecimen_ID) %>% distinct()
```

Reply:
> display the samples that have the ENSEMBL ID that are in PMTL list
> filtered_ensg %>% dplyr::filter(Ensembl_ID %in% pmtl_df$Ensembl_ID) %>% 
> select(Ensemble_ID, Kids_First_Biospecimen_ID) %>% distinct()

@runjin326, PR updated. For all the frequencies tables, removed ENSG IDs are not in the PMTL list.  `IGL-@, IGH@ IGH-@` genes (symbols) in the fusion tables are not displayed in the `ENSG-cohort` RMD rendered tables because they don't have associated `ENSG IDs`. But they are reported in the more comprehensive tables written to file, including `samples IDs`.

Reply:
for clarification - " removed ENSG IDs are not in the PMTL list" - or  is this removing ENSG IDs that are not in Ensemble pkg 104 and Gencode v38? or internal QC to identify any geneIDs that are in the PMTL list that would get removed due to the filter.

Reply:
I read the code - he mentioned `# Display samples in removed ENSEMBL IDs that are in PMTL list` and the code confirms it is an internal QC to identify any geneIDs that are in the PMTL list that would get removed due to the filter.